projects/rulenode-core-config/src/lib/components/action/azure-iot-hub-config.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component } from '@angular/core';
+import { AppState } from '@core/public-api';
+import { RuleNodeConfiguration, RuleNodeConfigurationComponent } from '@shared/public-api';
+import { Store } from '@ngrx/store';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { AzureIotHubCredentialsType, azureIotHubCredentialsTypes, azureIotHubCredentialsTypeTranslations } from '../../rulenode-core-config.models';
+
+@Component({
+  selector: 'tb-action-node-azure-iot-hub-config',
+  templateUrl: './azure-iot-hub-config.component.html',
+  styleUrls: ['./mqtt-config.component.scss']
+})
+export class AzureIotHubConfigComponent extends RuleNodeConfigurationComponent {
+
+  azureIotHubConfigForm: FormGroup;
+
+  allAzureIotHubCredentialsTypes = azureIotHubCredentialsTypes;
+  azureIotHubCredentialsTypeTranslationsMap = azureIotHubCredentialsTypeTranslations;
+
+  constructor(protected store: Store<AppState>,
+              private fb: FormBuilder) {
+    super(store);
+  }
+
+  protected configForm(): FormGroup {
+    return this.azureIotHubConfigForm;
+  }
+
+  protected onConfigurationSet(configuration: RuleNodeConfiguration) {
+    this.azureIotHubConfigForm = this.fb.group({
+      topicPattern: [configuration ? configuration.topicPattern : null, [Validators.required]],
+      host: [configuration ? configuration.host : null, [Validators.required]],
+      port: [configuration ? configuration.port : null, [Validators.required, Validators.min(1), Validators.max(65535)]],
+      connectTimeoutSec: [configuration ? configuration.connectTimeoutSec : null,
+        [Validators.required, Validators.min(1), Validators.max(200)]],
+      clientId: [configuration ? configuration.clientId : null, [Validators.required]],
+      cleanSession: [configuration ? configuration.cleanSession : false, []],
+      ssl: [configuration ? configuration.ssl : false, []],
+      credentials: this.fb.group(
+        {
+          type: [configuration && configuration.credentials ? configuration.credentials.type : null, [Validators.required]],
+          sasKey: [configuration && configuration.credentials ? configuration.credentials.sasKey : null, []],
+          caCert: [configuration && configuration.credentials ? configuration.credentials.caCert : null, []],
+          caCertFileName: [configuration && configuration.credentials ? configuration.credentials.caCertFileName : null, []],
+          privateKey: [configuration && configuration.credentials ? configuration.credentials.privateKey : null, []],
+          privateKeyFileName: [configuration && configuration.credentials ? configuration.credentials.privateKeyFileName : null, []],
+          cert: [configuration && configuration.credentials ? configuration.credentials.cert : null, []],
+          certFileName: [configuration && configuration.credentials ? configuration.credentials.certFileName : null, []]
```

Comment:
can be replaced `configuration && configuration.credentials` to `configuration?.credentials`